File: 274.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    int hIndex(vector<int>& citations) {
        int size=citations.size();
        vector<int> bucket(size+1,0);
        for(auto c:citations){
            if(c>=size)
                bucket[size]++;
            else
                bucket[c]++;
        }
        int sum=0;
        for(int i=size;i>=0;--i){
            sum+=bucket[i];
            if(i<=sum)
                return i;
        }
        return 0;
    }
};

int main(){

}

File: Python/274.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        if not citations:
            return 0
        n = len(citations)
        bucket = [0]*(n+1)
        for c in citations:
            if c >= n:
                bucket[n]+=1
            else:
                bucket[c] +=1
        count = 0
        for i in range(n, -1, -1):
            count += bucket[i]
            if count >= i:
                return i
        return 0
